### PR TITLE
br: deal the question client wait too long time(100 second) when the pd address is wrong or TLS is illegal

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close #https://github.com/pingcap/tidb/issues/27415

Problem Summary:

### What is changed and how it works?

What's Changed:
It get pd failed because for TLS, and it retried for 100 times(one second per one retry operation).
Now,
(1)change the retryCount from 100 to 3
(2)Create the pd client before opening the storage

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)

#### Before fix it
image

#### After fix it 
image

#### test backup/restore
image

```release-note
It need't wait too long when it met error in the operation of connecting pdClient 
```

### CODE REVIEW

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/conn/conn.go [hunk 1]
```diff
@@ -162,6 +163,27 @@ func GetAllTiKVStores(
...
+	}
+
+	liveStoreCount := 0
+	for _, s := range stores {
+		if s.GetState() != metapb.StoreState_Up {
+			continue
+		}
+		liveStoreCount++
+	}
+	log.Info("it has ", zap.Int("aliveStore", liveStoreCount))
```
Comment (kennytm) [inline br/pkg/conn/conn.go:183]:
```suggestion
	log.Info("checked alive KV stores", zap.Int("aliveStores", liveStoreCount), zap.Int("totalStores", len(stores)))
```

File: br/pkg/task/common.go [hunk 2]
```diff
@@ -398,15 +397,9 @@ func NewMgr(ctx context.Context,
 		}
 	}
 
-	// Disable GC because TiDB enables GC already.
```
Comment (kennytm) [inline br/pkg/task/common.go:401]: keep this comment in conn.go:228

**Review by kennytm — APPROVED** [review]

**Pull request merged**